enterprise/tpa_pipeline.py

```diff
@@ -142,10 +142,17 @@ def redirect_to_consent():
     if not enterprise_customer.requests_data_sharing_consent:
         # This enterprise customer attached to this pipeline element does not request data sharing consent;
         # proceed with the creation of a link between the user and the enterprise customer, then exit.
-        EnterpriseCustomerUser.objects.get_or_create(
+        ec_user, _ = EnterpriseCustomerUser.objects.get_or_create(
             enterprise_customer=enterprise_customer,
             user_id=user.id
         )
+
+        if enterprise_customer.enforce_data_sharing_consent == EnterpriseCustomer.EXTERNALLY_MANAGED:
```

Comment:
Yes, that is true - if consent is disabled we shouldn't write this record. I'm going to update this to make that clear, since I believe a model could have consent disabled but still have this value set to externally managed, and the consent enabled field takes precedence.